src/d3a/models/strategy/storage.py

```diff
@@ -15,208 +15,197 @@
+                                        initial_buying_rate=None, final_buying_rate=None,
+                                        energy_rate_change_per_update=None):
+        if battery_capacity_kWh is not None and battery_capacity_kWh < 0:
             raise ValueError("Battery capacity should be a positive integer")
-        if initial_soc and not min_allowed_soc <= initial_soc <= 100:
-            raise ValueError("Initial charge is a percentage value, should be between "
-                             "MIN_ALLOWED_SOC and 100.")
-        if risk and not 0 <= risk <= 100:
-            raise ValueError("Risk is a percentage value, should be between 0 and 100.")
-        if initial_capacity_kWh and battery_capacity_kWh and min_allowed_soc and \
-           not min_allowed_soc * battery_capacity_kWh <= \
-           initial_capacity_kWh <= battery_capacity_kWh:
-            raise ValueError(f"Initial capacity should be between min_allowed_capacity and "
-                             "battery_capacity_kWh parameter.")
-        if initial_selling_rate < 0:
+        if max_abs_battery_power_kW is not None and max_abs_battery_power_kW < 0:
+            raise ValueError("Battery Power rating must be a positive integer.")
+        if initial_soc is not None and 0 < initial_soc > 100:
+            raise ValueError("initial SOC must be in between 0-100 %")
+        if min_allowed_soc is not None and 0 < min_allowed_soc > 100:
+            raise ValueError("initial SOC must be in between 0-100 %")
+        if initial_soc is not None and min_allowed_soc is not None and \
+                initial_soc < min_allowed_soc:
+            raise ValueError("Initial charge must be more than the minimum allowed soc.")
+        if initial_selling_rate is not None and initial_selling_rate < 0:
             raise ValueError("Initial selling rate must be greater equal 0.")
+        if final_selling_rate is not None and final_selling_rate < 0:
+            raise ValueError("Final selling rate must be greater equal 0.")
+        if initial_selling_rate is not None and final_selling_rate is not None and \
+                initial_selling_rate < final_selling_rate:
+            raise ValueError("Initial selling rate must be greater than final selling rate.")
+        if initial_buying_rate is not None and initial_buying_rate < 0:
+            raise ValueError("Initial buying rate must be greater equal 0.")
+        if final_buying_rate is not None and final_buying_rate < 0:
+            raise ValueError("Final buying rate must be greater equal 0.")
+        if initial_buying_rate is not None and final_buying_rate is not None and \
+                initial_buying_rate > final_buying_rate:
+            raise ValueError("Initial buying rate must be less than final buying rate.")
+        if final_selling_rate is not None and final_buying_rate is not None and \
+                final_selling_rate < final_buying_rate:
```

Comment:
Cant they be equal as well? Then there is no revenue for the owner, but it should be possible right?
Also, please make the error message more verbose here. Thanks!